packages/amplify-category-function/src/provider-utils/awscloudformation/utils/packageLayer.ts

```diff
@@ -54,9 +53,10 @@ export async function packageLayer(context, resource) {
       reject(new Error('Failed to zip code.'));
     });
 
+    const libPath = path.join(resourcePath, 'lib', '*');
```

Comment:
I think you want '**' (to include contents of subdirectories as well)
And you may also want ** for the optPath as well

Reply:
The single '*' is intentional. If you use '**' the zipped folder would look like:
```
lib/nodejs/node_modules/
nodejs/node_modules/
node_modules/
opt/
```